extension.js

```diff
@@ -224,8 +225,11 @@ function focus_window_changed() {
 
     const win = global.display.focus_window;
     if (win !== null) {
-        if (current_window === win || current_window.is_ancestor_of_transient(win))
+        if (current_window === win || current_window.is_ancestor_of_transient(win)) {
+            if (settings.get_boolean('window-maximize'))
+                window_maximize();
```

Comment:
It's not the correct place for this. Maybe it accidentally works, but the purpose of this function is hiding the terminal if another application is activated (this behavior can be enabled/disabled in Preferences, off by default).

I think initial `maximize()` should be added at the end of [`track_window()`](https://github.com/amezin/gnome-shell-extension-ddterm/blob/1c9af7165a94e880d5196cea6306051321363dea/extension.js#L311) (where new terminal window is detected and configured). Another possible location is [`update_window_geometry()`](https://github.com/amezin/gnome-shell-extension-ddterm/blob/1c9af7165a94e880d5196cea6306051321363dea/extension.js#L371) (i. e. unmaximize if not `window-maximize`, but also ensure the window is maximized if `window-maximize`)